process/smartContract/builtInFunctions/keyValueStorage.go

```diff
@@ -13,17 +13,45 @@ import (
 
 var _ process.BuiltinFunction = (*saveKeyValueStorage)(nil)
 
+const numInitCharactersInProtectedKeys = 4
+
 type saveKeyValueStorage struct {
-	gasConfig   BaseOperationCost
-	funcGasCost uint64
+	gasConfig      BaseOperationCost
+	funcGasCost    uint64
+	mapInvalidKeys map[string]struct{}
 }
 
 // NewSaveKeyValueStorageFunc returns the key-value
 func NewSaveKeyValueStorageFunc(
 	gasConfig BaseOperationCost,
 	funcGasCost uint64,
-) *saveKeyValueStorage {
-	return &saveKeyValueStorage{gasConfig: gasConfig, funcGasCost: funcGasCost}
+	mapInvalidKeys map[string]struct{},
+) (*saveKeyValueStorage, error) {
+	if mapInvalidKeys == nil {
+		return nil, process.ErrNilInvalidKeysMap
+	}
+
+	s := &saveKeyValueStorage{
+		gasConfig:   gasConfig,
+		funcGasCost: funcGasCost,
+	}
+	s.mapInvalidKeys = make(map[string]struct{}, len(mapInvalidKeys))
+	for key := range mapInvalidKeys {
+		trimmedKey := key[:numInitCharactersInProtectedKeys]
```

Comment:
If len of key is lower than numInitCharactersInProtectedKeys the code would not panicked here?

Reply:
no, it would not panic - it goes to the index only if exist.

Reply:
ok

Reply:
yes it panics, added check.